src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -171,7 +173,21 @@ public NSUrlSessionHandler (NSUrlSessionConfiguration configuration)
 			inflightRequests = new Dictionary<NSUrlSessionTask, InflightData> ();
 		}
 
-#if !MONOMAC
+#if !MONOMAC && !WATCH
+
+		void AddNotification ()
+		{
+			if (!isBackgroundSession && notificationToken == null)
+				using (var notification = new NSString ("UIApplicationWillResignActiveNotification"))
+					notificationToken = NSNotificationCenter.DefaultCenter.AddObserver (notification, BackgroundNotificationCb);
+		}
+
+		void RemoveNotification ()
+		{
+			if (notificationToken != null)
+				NSNotificationCenter.DefaultCenter.RemoveObserver (notificationToken);
```

Comment:
You should set `notificationToken` to null so that it can be re-added if more requests come in.

Reply:
God.. what a stupid mistake! thx!